### ISSUE TITLE
Synchronous calls to server?

### ISSUE THREAD

**Issue by PhilAndrew** [memgraph/mgclient#1]
Hi, this looks like its synchronous blocking functions am I right? Then if I wish to make many calls, can I 
1) Do them in parallel threads, is it thread safe?
2) Should I be doing them in threads, what if I want to run 1000 calls at the same time, 1000 threads?

**Comment by tomicm** [comment]
Yes, query functions are synchronous and blocking. If you wish to make many calls in parallel, you can make them from parallel threads. However, you must create a `mg_session` object per thread, since the objects aren't thread safe, and you must pull all the results before making the next call.

Since you need a separate connection for each `mg_session` object, the number of calls that you can make in parallel is limited by the maximum number of connections the server is configured to accept.
